exercises/binary-search-tree/binary_search_tree_test.cpp

```diff
@@ -88,38 +88,42 @@ BOOST_AUTO_TEST_CASE(can_create_complex_tree)
 template<typename T>
 static void test_sort(const tree_ptr<T> &tree, const std::vector<T> &expected)
 {
-    BOOST_REQUIRE_EQUAL_COLLECTIONS(tree->begin(), tree->end(), expected.begin(), expected.end());
+    std::vector<T> actual;
```

Comment:
I think it will be clearer if we write:
```cpp
std::vector<T> actual(tree->begin(), tree->end);
REQUIRE(expected == actual);
```
This way it's clear what functions you need to implement.

Reply:
Hmm, my thinking was they're going to have to look it up anyways, so why not show them you can build something that nicely uses the language features?

You make the call on this one and I'll switch it to your preference.

Reply:
Agree. Lets start like that, and if students complain we can change it to be more explicit.